src/Cubie/Moves/Edges.hs

```diff
@@ -0,0 +1,73 @@
+module Cubie.Moves.Edges
+( front
+, back
+, left
+, right
+, up
+, down
+) where
+
+import qualified Cubie.Moves as M (accumulate, permute)
+
+-- set the bound on the accumulator
+accumulate = M.accumulate (2 :: Int) -- TODO: how do i fix type declarations so this cast isnt needed?
```

Comment:
@SaxJ I'm still struggling with the type system. If you are able to, can you pull this branch and tell me what I'm doing wrong? If this type cast is removed the source won't compile and I'm not sure how else to fix it (unless this is the _correct_ way?)